graphql/schema.lua

```diff
@@ -99,6 +99,26 @@ end
 function schema:generateDirectiveMap()
   for _, directive in ipairs(self.directives) do
     self.directiveMap[directive.name] = directive
+    if directive.arguments then
+      for name, argument in pairs(directive.arguments) do
+
+          -- BEGIN_HACK: resolve type names to real types
+          if type(argument) == 'string' then
+            argument = types.resolve(argument, self.name)
+            directive.arguments[name] = argument
+          end
+
+          if type(argument.kind) == 'string' then
+            argument.kind = types.resolve(argument.kind, self.name)
+          end
+          -- END_HACK: resolve type names to real types
+
+          local argumentType = argument.__type and argument or argument.kind
+          assert(argumentType, 'Must supply type for argument "' .. name .. '" on "' .. directive.name .. '"')
```

Comment:
Assert with concatenation looks bad. It's performed even assert isn't triggered. Pease change to `if ... then error() end` if your really need it.

Reply:
Are u shure? I've made this exactly the same way as in other places of this module. You have plans to refactor all the module?

Reply:
I'm completely sure. You can easily prove it using memprof.
I don't have plans to refactor this module. For now we could only write optimal code and probably fix other places one day.